docs/sql-data-sources-csv.md

```diff
@@ -38,3 +38,222 @@ Spark SQL provides `spark.read().csv("file_name")` to read a file or directory o
 </div>
 
 </div>
+
+## Data Source Option
+
+Data source options of CSV can be set via:
+* the `.option`/`.options` methods of
+  *  `DataFrameReader`
+  *  `DataFrameWriter`
+  *  `DataStreamReader`
+  *  `DataStreamWriter`
+  *  `from_csv`
+  *  `to_csv`
+  *  `schema_of_csv`
+  *  `OPTIONS` clause at [CREATE TABLE USING DATA_SOURCE](sql-ref-syntax-ddl-create-table-datasource.html)
+
+
+<table class="table">
+  <tr><th><b>Property Name</b></th><th><b>Default</b></th><th><b>Meaning</b></th><th><b>Scope</b></th></tr>
+  <tr>
+    <td><code>sep</code></td>
+    <td>,</td>
+    <td>Sets a separator (one or more characters) for each field and value. If None is set, it uses the default value, <code>,</code>.</td>
+    <td>read/write</td>
+  </tr>
+  <tr>
+    <td><code>encoding</code></td>
+    <td><code>UTF-8</code> for reading, not set for writing</td>
+    <td>For reading, decodes the CSV files by the given encoding type. If None is set, it uses the default value, <code>UTF-8</code>. For writing, sets the encoding (charset) of saved csv files. If None is set, the default UTF-8 charset will be used.</td>
+    <td>read/write</td>
+  </tr>
+  <tr>
+    <td><code>quote</code></td>
+    <td>"</td>
+    <td>Sets a single character used for escaping quoted values where the separator can be part of the value. If None is set, it uses the default value, <code>"</code>. If you would like to turn off quotations, you need to set an empty string. If an empty string is set, it uses <code>u0000</code> (null character).</td>
+    <td>read/write</td>
+  </tr>
+  <tr>
+    <td><code>quoteAll</code></td>
+    <td>false</td>
+    <td>A flag indicating whether all values should always be enclosed in quotes. If None is set, it uses the default value <code>false</code>, only escaping values containing a quote character.</td>
+    <td>write</td>
+  </tr>
+  <tr>
+    <td><code>escape</code></td>
+    <td>\</td>
+    <td>Sets a single character used for escaping quotes inside an already quoted value. If None is set, it uses the default value, <code>\</code>.</td>
+    <td>read/write</td>
+  </tr>
+  <tr>
+    <td><code>escapeQuotes</code></td>
+    <td>true</td>
+    <td>a flag indicating whether values containing quotes should always be enclosed in quotes. If None is set, it uses the default value <code>true</code>, escaping all values containing a quote character.</td>
+    <td>write</td>
+  </tr>
+  <tr>
+    <td><code>comment</code></td>
+    <td>""</td>
+    <td>Sets a single character used for skipping lines beginning with this character. By default (None), it is disabled.</td>
```

Comment:
Yeah, I think we can just remove the sentences such as "By default ..." or "If None is set ...", since now we have proper default value in the `Default` column, so it's duplicated.

Let me adress this